Lib/fontTools/merge.py

```diff
@@ -919,21 +988,63 @@ def __init__(self, options=None):
 		self.options = options
 
 	def merge(self, fontfiles):
-
-		mega = ttLib.TTFont()
+		# Take first input file sfntVersion
+		sfntVersion = ttLib.TTFont(fontfiles[0]).sfntVersion
+		mega = ttLib.TTFont(sfntVersion=sfntVersion)
 
 		#
 		# Settle on a mega glyph order.
 		#
 		fonts = [ttLib.TTFont(fontfile) for fontfile in fontfiles]
+
+		
+		# Check if all fonts have CFF
+		allOTFs = all(cff is not None for cff in [font.get('CFF ') for font in fonts])
+
+		cffTables = []
+		if allOTFs:
+			for font in fonts:
+				# Desubroutinize with Subsetter (like compreffor does)
+				from io import BytesIO
+
+				stream = BytesIO()
+				font.save(stream, reorderTables=False)
+				stream.flush()
+				stream.seek(0)
+				tmpfont = ttLib.TTFont(stream)
+
+				options = subset.Options()
+				options.desubroutinize = True
+				options.notdef_outline = True
+				subsetter = subset.Subsetter(options=options)
+				subsetter.populate(glyphs=tmpfont.getGlyphOrder())
+				subsetter.subset(tmpfont)
+				data = tmpfont['CFF '].compile(tmpfont)
+				table = ttLib.newTable('CFF ')
+				table.decompile(data, font)
+				font['CFF '] = table
+				cffTables.append(font.get('CFF '))
+				tmpfont.close()
+
+
 		glyphOrders = [font.getGlyphOrder() for font in fonts]
-		megaGlyphOrder = self._mergeGlyphOrders(glyphOrders)
+		
+		# Handle glyphOrder merging and cff glyphs renaming together
+		megaGlyphOrder = self._mergeGlyphOrders(glyphOrders, cffTables)
+
 		# Reload fonts and set new glyph names on them.
 		# TODO Is it necessary to reload font?  I think it is.  At least
 		# it's safer, in case tables were loaded to provide glyph names.
-		fonts = [ttLib.TTFont(fontfile) for fontfile in fontfiles]
-		for font,glyphOrder in zip(fonts, glyphOrders):
-			font.setGlyphOrder(glyphOrder)
+		# fonts = [ttLib.TTFont(fontfile) for fontfile in fontfiles]
+
+		if len(cffTables):
+			for font, glyphOrder, cffTable in zip(fonts, glyphOrders, cffTables):
+				font.setGlyphOrder(glyphOrder)
+				font['CFF '] = cffTable
```

Comment:
why are you setting the CFF tables again in this line? You already set them to the same same objects a few lines above, no?
https://github.com/fonttools/fonttools/pull/1289/files#diff-4ac66bcb14ce18ebedd3df81c04259deR1025

Reply:
that's because the fonts 'should' be reloaded… (see commented line 1038)